app/components/bulk_upload_error_row_component.html.erb

```diff
@@ -9,24 +9,27 @@
 
   <div class="govuk-summary-card__content">
     <% potential_errors, critical_errors = bulk_upload_errors.partition { |error| error.category == "soft_validation" } %>
-    <h2 class="govuk-heading-m">Critical errors</h2>
-    <p class="govuk-body">These errors must be fixed to complete your logs.</p>
-    <%= govuk_table do |table| %>
-      <%= table.with_head do |head| %>
-        <% head.with_row do |row| %>
-          <% row.with_cell(header: true, text: "Cell") %>
-          <% row.with_cell(header: true, text: "Question") %>
-          <% row.with_cell(header: true, text: "Error") %>
-          <% row.with_cell(header: true, text: "Specification") %>
-        <% end %>
 
-        <%= table.with_body do |body| %>
-          <% critical_errors.each do |error| %>
-            <% body.with_row do |row| %>
-                <% row.with_cell(text: error.cell) %>
-                <% row.with_cell(text: question_for_field(error.field), html_attributes: { class: "govuk-!-width-one-half" }) %>
-                <% row.with_cell(text: error.error.html_safe, html_attributes: { class: "govuk-!-font-weight-bold govuk-!-width-one-half" }) %>
-                <% row.with_cell(text: error.field.humanize) %>
+    <% if critical_errors.any? %>
+      <h2 class="govuk-heading-m">Critical errors</h2>
+      <p class="govuk-body">These errors must be fixed to complete your logs.</p>
+      <%= govuk_table do |table| %>
+        <%= table.with_head do |head| %>
+          <% head.with_row do |row| %>
+            <% row.with_cell(header: true, text: "Cell") %>
+            <% row.with_cell(header: true, text: "Question") %>
+            <% row.with_cell(header: true, text: "Error") %>
+            <% row.with_cell(header: true, text: "Specification") %>
+          <% end %>
+
+          <%= table.with_body do |body| %>
+            <% critical_errors.each do |error| %>
+              <% body.with_row do |row| %>
+                  <% row.with_cell(text: error.cell) %>
+                  <% row.with_cell(text: question_for_field(error.field), html_attributes: { class: "govuk-!-width-one-half" }) %>
+                  <% row.with_cell(text: error.error.html_safe, html_attributes: { class: "govuk-!-font-weight-bold govuk-!-width-one-half" }) %>
+                  <% row.with_cell(text: error.field.humanize) %>
+              <% end %>
```

Comment:
In the bulk upload full error report, we now make showing the critical errors table also conditional on if there are any.